### PR TITLE
Move webhook destroy job from ActiveJob to Sidekiq

### PR DESCRIPTION
**Pull request by citizen428**
## What type of PR is this? (check all applicable)

- [X] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Moved `Webhook::DestroyJob` to `Webhook::DestroyWorker`. 

## Related Tickets & Documents

#5305

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [X] no documentation needed

### CODE REVIEW

File: app/controllers/oauth/tokens_controller.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ def revoke
       # the access token model.
       if authorized?
         revoke_token
-        Webhook::DestroyJob.perform_later(user_id: token.resource_owner_id, application_id: token.application_id)
+        Webhook::DestroyWorker.perform_async(token.resource_owner_id, token.application_id)
```
Comment (citizen428) [inline app/controllers/oauth/tokens_controller.rb]: Note: I removed the keyword args for two reasons:

1. It's more consistent with our other jobs.
2. They don't play well with the new `sidekiq_perform_enqueued_jobs`. 

Re 2:

At the workers, the args arrive as a single hash argument and are not splatted into kwargs:

```
[1] dev.to(SidekiqTestHelpers::Utils)> job
{
         "args" => [
    [0] {
      "application_id" => 50,
             "user_id" => 181
    }
  ],
        "class" => "Webhook::DestroyWorker",
   "created_at" => 1578636506.831237,
  "enqueued_at" => 1578636506.831522,
          "jid" => "013c3accfdc8a4b9d48c1ba5",
        "queue" => "low_priority",
        "retry" => 10
}
```

This leads to the following error:

```
ArgumentError:
  wrong number of arguments (given 1, expected 0; required keywords: user_id, application_id)
```

It's something we may want to address in the future, but it's not a rabbit hole I did want to go down right now.

Comment (rhymes) [reply app/controllers/oauth/tokens_controller.rb]: Keyword arguments are also badly supported by Sidekiq: https://github.com/mperham/sidekiq/wiki/Best-Practices#1-make-your-job-parameters-small-and-simple

> Don't pass symbols, named parameters or complex Ruby objects (like Date or Time!) as those will not survive the dump/load round trip correctly.

good call to remove them

Comment (citizen428) [reply app/controllers/oauth/tokens_controller.rb]: That’s exactly the reason for 2. mentioned above. The kwargs don’t survive the round trip to Sidekiq but come back as hash instead.

**Comment by mstruve** [comment]
Closing in favor of https://github.com/thepracticaldev/dev.to/pull/5424

**Pull request closed without merging**